code/modules/shuttles/shuttle.dm

```diff
@@ -253,14 +279,17 @@
 			var/datum/powernet/NewPN = new()
 			NewPN.add_cable(C)
 			propagate_network(C,C.powernet)
-
-	if(mothershuttle)
-		var/datum/shuttle/mothership = SSshuttle.shuttles[mothershuttle]
-		if(mothership)
-			if(current_location.landmark_tag == motherdock)
-				mothership.shuttle_area |= shuttle_area
-			else
-				mothership.shuttle_area -= shuttle_area
+	for(var/datum/pipe_network/pipenet in pipenets)
```

Comment:
Can you reply `powernet` and `return_network()` to return correct instances? If so you could `as anything` these loops.